### ISSUE TITLE
2 aircraft at the same time land and take off in a neverending loop

### ISSUE THREAD

**Issue by Mailaender** [OpenRA/ra2#683]
![ra2-aircraft-updown](https://user-images.githubusercontent.com/756669/77229551-111f9800-6b8f-11ea-915b-45e13e218037.gif)

**Comment by sorcerer86pt** [comment]
i saw this many times. It's because the logic for multiple points of landing on the building is missed. It should be calculated for 4, but it only uses the first one.

**Comment by Phrohdoh** [comment]
#735 seems like a mod-level workaround (that I worry we'll end up having to maintain when we don't actually need it) to the engine not supporting multiple landing/takeoff points.

Was it decided that downstream was the proper place to implement this, instead of in the engine?

**Comment by abcdefg30** [comment]
> Was it decided that downstream was the proper place to implement this, instead of in the engine?

Not exactly, #735 is afaik ported from RV which implemented this workaround to not need to wait for the engine's docking code to be rewritten.

**Comment by Phrohdoh** [comment]
> engine's docking code to be rewritten
Seems like the proper fix that we should pursue, if we're going to do anything here.  IME workarounds stay in the codebase even after a proper implementation is available, and if something is a pile of workarounds the maintenance burden grows higher and pool of people willing to work on it diminishes (to potentially even less people than now).
